src/Gallery.js

```diff
@@ -0,0 +1,146 @@
+import React from "react";
+
+const STATUS_FETCHING = "fetching";
+const STATUS_LOADED = "loaded";
+
+export default class Gallery extends React.Component {
+  state = {
+      loadingState: null,
+      images: [],
+      page: 0,
+      loadLimit: 10,
+      limitImagesOnPage: 1500,
+      isScrollbarVisible: false,
+      isMobile: false,
+  };
+
+  componentDidMount() {
+    const loadImagesAtStart = () => {
+      if (window.innerHeight > document.body.offsetHeight) {
+        this.fetchRandomCat();
+      } 
+      if (window.innerHeight < document.body.offsetHeight) {
+        clearInterval(interval);
+        this.setState({isScrollbarVisible: true});
+      }
+    }
+    const interval = setInterval(loadImagesAtStart, 500);
+    this.screenWidth();
+    window.addEventListener('scroll', this.handleScroll);
+    window.addEventListener('resize', this.screenWidth);
+  }
+
+  componentWillUnmount() {
+    window.removeEventListener("scroll", this.handleScroll);
+    window.removeEventListener("resize", this.screenWidth); 
+  }
+
+  handleScroll = () => {
+    if ( (window.scrollY + window.innerHeight) >= document.body.offsetHeight ) {
+      if (this.state.loadingState !== STATUS_FETCHING) {
+        this.fetchRandomCat();
+      } 
+    }
+  }
+
+  screenWidth = () => {
+    if (window.innerWidth < 600 && !this.state.isMobile) {
+      this.setState({isMobile: true});
+    }
+    if (window.innerWidth > 600 && this.state.isMobile) {
+      this.setState({isMobile: false});
+    }
+  }
+
+  autoScrollToBottom = () => {
+    const date = new Date();
+    const scrollingTime = 10000;
+    const endTime = date.getTime() + scrollingTime;
+    const startScrolling = (images = this.loadedImages, limitImages = this.state.limitImagesOnPage) => {
+      // if scrollbar is not visible, then load more images; if is visible, then scroll to bottom, which fires loading more images
+      if (!this.state.isScrollbarVisible) {
+        this.fetchRandomCat();
+      } 
+      window.scrollTo({
+        top: document.body.offsetHeight,
+        behavior: 'smooth'
+      });
+      // clear interval
+      const newDate = new Date();
+      const currentTime = newDate.getTime();
+      if (currentTime >= endTime) {
+        clearInterval(interval);
+        window.scrollTo({
+          top: (this.galleryContainer.offsetTop + this.galleryContainer.getBoundingClientRect().height) - window.innerHeight,
```

Comment:
type error: po wlaczeniu autoscrolla i zmianie zakladki na na random nie jest sprawdzone czy this.galleryContainer nie jest nullem